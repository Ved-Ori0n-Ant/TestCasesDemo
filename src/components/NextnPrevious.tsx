import React from 'react';
import {Button, StyleSheet, View} from 'react-native';
import {RootStackParamList} from '../navigation/MainStack';
import {StackActions, useNavigation} from '@react-navigation/native';
import {NativeStackNavigationProp} from '@react-navigation/native-stack';

type props = {
  currentScreen: Stack;
  testID?: string;
};
const SCREENS = [
  'TextScreen',
  'TextTranslationScreen',
  'TextInputScreen',
  'ButtonScreen',
  'ImageScreen',
  'FlatListScreen',
];
export enum Stack {
    'TextScreen',
    'TextTranslationScreen',
    'TextInputScreen',
    'ButtonScreen',
    'ImageScreen',
    'FlatListScreen',
}

const NextnPrevious: React.FC<props> = ({currentScreen, testID}) => {
  const navigation =
    useNavigation<NativeStackNavigationProp<RootStackParamList>>();

  const onPressNext = () => {
    navigation.dispatch(StackActions.replace(SCREENS[currentScreen + 1]));
  };
  const onPressPrevious = () => {
    navigation.dispatch(StackActions.replace(SCREENS[currentScreen - 1]));
  };
  return (
    <View style={styles.body} testID={testID || 'np_btn'}>
      <Button
        title={'Previous'}
        testID={'Previous_btn'}
        onPress={onPressPrevious}
        disabled={currentScreen <= 0}
      />
      <Button
        title={'Next'}
        testID={'Next_btn'}
        onPress={onPressNext}
        disabled={currentScreen >= SCREENS.length - 1}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  body: {
    backgroundColor: '#ddd',
    flexDirection: 'row',
    justifyContent: 'space-evenly',
  },
});

export default NextnPrevious;
