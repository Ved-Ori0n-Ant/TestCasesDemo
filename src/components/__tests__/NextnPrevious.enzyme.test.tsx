import React from "react";
import NextnPrevious, { Stack } from "../NextnPrevious";
import { shallow } from "enzyme";
import { NavigationContainer } from "@react-navigation/native";

const mockDispatch = jest.fn();
jest.mock('@react-navigation/native', () => ({
  ...jest.requireActual('@react-navigation/native'),
  useNavigation: () => ({
    dispatch: mockDispatch,
  }),
}));

describe('tests for NextnPrevios screen', () => {
    it('should render okay', () => {
        const tree = shallow(<NextnPrevious currentScreen={Stack.TextScreen} />)
        expect(tree).toMatchSnapshot();
    });
    it('should have two buttons', () => {
        const tree = shallow(<NavigationContainer><NextnPrevious currentScreen={Stack.TextScreen} /></NavigationContainer>)
        const previousBtn = tree.findWhere(node => node.prop('testID') === 'Previous_btn')
        const nextBtn = tree.findWhere(node => node.prop('testID') === 'Next_btn')

        expect(previousBtn).not.toBeNull();
        expect(nextBtn).not.toBeNull();
        expect(previousBtn).toBeDefined();
        expect(nextBtn).toBeDefined();
    });
    it('should render disable previousBtn while nextBtn should be active', async() => {
        const tree = shallow(<NextnPrevious currentScreen={Stack.TextScreen} />)
        const previousBtn = tree.findWhere(node => node.prop('testID') === 'Previous_btn')  
        const nextBtn = tree.findWhere(node => node.prop('testID') === 'Next_btn')
        expect(nextBtn.props().disabled).toBe(false);
        expect(previousBtn.props().disabled).toBe(true);
    });
    it('should render disable nextBtn while prevBtn should be active', async() => {
      const tree = shallow(<NextnPrevious currentScreen={Stack.FlatListScreen} />)
      const previousBtn = tree.findWhere(node => node.prop('testID') === 'Previous_btn')  
      const nextBtn = tree.findWhere(node => node.prop('testID') === 'Next_btn')
      expect(previousBtn.props().disabled).toBe(false);
      expect(nextBtn.props().disabled).toBe(true)
    });
    it('should call navigate method', async() => {
      const tree = shallow(<NextnPrevious currentScreen={Stack.ImageScreen} />)
      const previousBtn = tree.findWhere(node => node.prop('testID') === 'Previous_btn')  
      const nextBtn = tree.findWhere(node => node.prop('testID') === 'Next_btn')
      
      previousBtn.props().onPress();
      expect(mockDispatch).toBeCalledTimes(1)

      nextBtn.props().onPress();
      expect(mockDispatch).toBeCalledTimes(2)
    })
})