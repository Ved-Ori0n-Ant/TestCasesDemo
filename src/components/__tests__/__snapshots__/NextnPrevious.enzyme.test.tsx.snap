// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for NextnPrevios screen should render okay 1`] = `
<View
  style={
    {
      "backgroundColor": "#ddd",
      "flexDirection": "row",
      "justifyContent": "space-evenly",
    }
  }
  testID="np_btn"
>
  <Button
    disabled={true}
    onPress={[Function]}
    testID="Previous_btn"
    title="Previous"
  />
  <Button
    disabled={false}
    onPress={[Function]}
    testID="Next_btn"
    title="Next"
  />
</View>
`;
